utils/frame/remote-externalities/src/lib.rs

```diff
@@ -113,33 +113,45 @@ use sp_core::{
 	storage::{StorageKey, StorageData},
 };
 use codec::{Encode, Decode};
-use jsonrpsee_http_client::{HttpClient, HttpClientBuilder};
-
 use sp_runtime::traits::Block as BlockT;
+use jsonrpsee_ws_client::{WsClientBuilder, WsClient};
 
 type KeyPair = (StorageKey, StorageData);
 
 const LOG_TARGET: &str = "remote-ext";
-const TARGET: &str = "http://localhost:9933";
+const DEFAULT_TARGET: &str = "wss://rpc.polkadot.io";
 
 jsonrpsee_proc_macros::rpc_client_api! {
 	RpcApi<B: BlockT> {
-		#[rpc(method = "state_getPairs", positional_params)]
-		fn storage_pairs(prefix: StorageKey, hash: Option<B::Hash>) -> Vec<(StorageKey, StorageData)>;
-		#[rpc(method = "chain_getFinalizedHead")]
+		#[rpc(method = "state_getStorage", positional_params)]
+		fn get_storage(prefix: StorageKey, hash: Option<B::Hash>) -> StorageData;
+		#[rpc(method = "state_getKeysPaged", positional_params)]
+		fn get_keys_paged(
+			prefix: Option<StorageKey>,
+			count: u32,
+			start_key: Option<StorageKey>,
+			hash: Option<B::Hash>,
+		) -> Vec<StorageKey>;
+		#[rpc(method = "chain_getFinalizedHead", positional_params)]
 		fn finalized_head() -> B::Hash;
 	}
 }
 
 /// The execution mode.
 #[derive(Clone)]
-pub enum Mode<B: BlockT> {
+pub enum Mode<'a, B: BlockT> {
 	/// Online.
-	Online(OnlineConfig<B>),
+	Online(OnlineConfig<'a, B>),
 	/// Offline. Uses a state snapshot file and needs not any client config.
 	Offline(OfflineConfig),
 }
 
+impl<B: BlockT> Default for Mode<'_, B> {
```

Comment:
derive `Default` doesn't work?

Reply:
1. Enums can't derive default, right? 
2. Will hit the same wall as mentioned here:

>  // NOTE: ideally we would use `DefaultNoBound` here, but not worth bringing in frame-support for that.

Essentially, Derive will force us to bound `B: Default` as well, which is redundant.

Reply:
> Enums can't derive default, right?

yes, dumb question :)